gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreReflogEntry.java

```diff
@@ -1,25 +1,51 @@
 package com.virtuslab.gitcore.impl.jgit;
 
+import java.time.Instant;
+
 import io.vavr.control.Option;
 import lombok.AccessLevel;
-import lombok.Data;
 import lombok.RequiredArgsConstructor;
+import lombok.ToString;
+import org.checkerframework.checker.interning.qual.UsesObjectEquals;
 import org.eclipse.jgit.lib.ReflogEntry;
 
+import com.virtuslab.gitcore.api.IGitCoreCheckoutEntry;
 import com.virtuslab.gitcore.api.IGitCoreCommitHash;
 import com.virtuslab.gitcore.api.IGitCoreReflogEntry;
 
-@Data
-@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
+@RequiredArgsConstructor(access = AccessLevel.MODULE)
+@ToString(onlyExplicitlyIncluded = true)
+@UsesObjectEquals
 public class GitCoreReflogEntry implements IGitCoreReflogEntry {
-  private final String comment;
-  private final Option<IGitCoreCommitHash> oldCommitHash;
-  private final IGitCoreCommitHash newCommitHash;
-
-  static GitCoreReflogEntry of(ReflogEntry reflogEntry) {
-    return new GitCoreReflogEntry(
-        reflogEntry.getComment(),
-        GitCoreCommitHash.ofZeroable(reflogEntry.getOldId()),
-        GitCoreCommitHash.of(reflogEntry.getNewId()));
+
+  private final ReflogEntry reflogEntry;
+
+  @Override
+  @ToString.Include(name = "comment")
+  public String getComment() {
+    return reflogEntry.getComment();
+  }
+
+  @Override
+  @ToString.Include(name = "timestamp")
+  public Instant getTimestamp() {
+    return reflogEntry.getWho().getWhen().toInstant();
```

Comment:
Jeez, I was about to patch the JGit codebase just b/c I thought they don't expose the reflog timestamps at all... design over 9000

Reply:
Yeah, all those 3 invoked methods are rather lightweight (no access to the underlying repository or any other blocking operation) and can be reasonably expected to be deterministic wrt. returned value, so I think it can stay `get...` rather than `derive...`.